specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2024-04-01-preview/AlertRules.json

```diff
@@ -0,0 +1,2600 @@
+      "x-ms-client-flatten": true
+    },
+    "NrtAlertRuleProperties": {
+      "description": "Nrt alert rule base property bag.",
+      "properties": {
+        "alertRuleTemplateName": {
+          "description": "The Name of the alert rule template used to create this rule.",
+          "type": "string"
+        },
+        "templateVersion": {
+          "description": "The version of the alert rule template used to create this rule - in format <a.b.c>, where all are numbers, for example 0 <1.0.2>",
+          "type": "string"
+        },
+        "description": {
+          "description": "The description of the alert rule.",
+          "type": "string"
+        },
+        "query": {
+          "description": "The query that creates alerts for this rule.",
+          "type": "string"
+        },
+        "tactics": {
+          "description": "The tactics of the alert rule",
+          "items": {
+            "$ref": "./common/AlertTypes.json#/definitions/AttackTactic"
+          },
+          "type": "array"
+        },
+        "techniques": {
+          "description": "The techniques of the alert rule",
+          "items": {
+            "type": "string"
+          },
+          "type": "array"
+        },
+        "subTechniques": {
+          "description": "The sub-techniques of the alert rule",
+          "items": {
+            "type": "string"
+          },
+          "type": "array"
+        },
+        "displayName": {
+          "description": "The display name for alerts created by this alert rule.",
+          "type": "string"
+        },
+        "enabled": {
+          "description": "Determines whether this alert rule is enabled or disabled.",
+          "type": "boolean"
+        },
+        "lastModifiedUtc": {
+          "description": "The last time that this alert rule has been modified.",
+          "format": "date-time",
+          "readOnly": true,
+          "type": "string"
+        },
+        "suppressionDuration": {
+          "description": "The suppression (in ISO 8601 duration format) to wait since last time this alert rule been triggered.",
+          "format": "duration",
+          "type": "string"
+        },
+        "suppressionEnabled": {
+          "description": "Determines whether the suppression for this alert rule is enabled or disabled.",
+          "type": "boolean"
+        },
+        "severity": {
+          "$ref": "./common/AlertTypes.json#/definitions/AlertSeverityEnum",
+          "description": "The severity for alerts created by this alert rule."
+        },
+        "incidentConfiguration": {
+          "$ref": "#/definitions/IncidentConfiguration",
+          "description": "The settings of the incidents that created from alerts triggered by this analytics rule"
+        },
+        "customDetails": {
+          "description": "Dictionary of string key-value pairs of columns to be attached to the alert",
+          "type": "object",
+          "additionalProperties": {
```

Comment:
These properties are variable and frequently change, with each request potentially having different values for each entity. They were created to accommodate the need for a dynamically changing dictionary, and the data derived from it is utilized across various scenarios.